configtemplate/metadata/pivnet.go

```diff
@@ -177,28 +181,24 @@ func productFileKeysByGlobs(
 
 type AuthenticatedPivnetClient struct {
 	ClientConfig pivnetapi.ClientConfig
+	TokenService pivnetapi.AccessTokenService
 	HTTPClient   *http.Client
 }
 
 func (a AuthenticatedPivnetClient) Do(req *http.Request) (*http.Response, error) {
-	const legacyAPITokenLength = 20
-	if len(a.ClientConfig.Token) > legacyAPITokenLength {
-		baseURL := fmt.Sprintf("%s%s", a.ClientConfig.Host, "/api/v2")
-		tokenFetcher := pivnetapi.NewTokenFetcher(baseURL, a.ClientConfig.Token)
-		var err error
-		accessToken, err := tokenFetcher.GetToken()
-
-		if err != nil {
-			log.Fatalf("Exiting with error: %s", err)
-		}
-		req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", accessToken))
-	} else {
-		req.Header.Add("Authorization", fmt.Sprintf("Token %s", a.ClientConfig.Token))
+	token, err := a.TokenService.AccessToken()
```

Comment:
main difference is, we now need to create a TokenService which is in charge of dealing with the AccessToken.